pulp_rpm/app/modulemd.py

```diff
@@ -71,11 +71,10 @@ def split_modulemd_file(file):
             for line in modulemd_file:
                 if line.startswith("---"):
                     if module:  # avoid first empty yield in the beginning of document
-                        yield module
+                        yield module.rstrip("\n")
                         module = ""
                 module += line
-
-            yield module
```

Comment:
TL;DR if the module file didn't have a trailing newline, then the last module parsed wouldn't either, resulting in `...---` if they got published in a different order.

Solution, normalize the blocks to never contain newlines at the ends and insert them ourselves.